src/emulator/hardware/builtin-gates/MipsAlu.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * The MIT License (MIT)
+ * Copyright (c) 2017-present Dmitry Soshnikov <dmitry.soshnikov@gmail.com>
+ */
+
+'use strict';
+
+const BuiltInGate = require('../BuiltInGate');
+
+/**
+ * Canonical truth table for the MipsAlu
+ */
+const TRUTH_TABLE = [
+
+  // Add
+  {a: 1, b: 1, na: 0, nb: 0, less: 0, cin: 0, op: 0b10, out: 0, cout: 1, set: 0 },
+  {a: 1, b: 1, na: 0, nb: 0, less: 0, cin: 1, op: 0b10, out: 1, cout: 1, set: 0 },
+  {a: 0, b: 0, na: 0, nb: 0, less: 0, cin: 1, op: 0b10, out: 1, cout: 0, set: 0 },
+
+  // Sub
+  {a: 1, b: 1, na: 0, nb: 1, less: 0, cin: 1, op: 0b10, out: 0, cout: 1, set: 0 },
+
+  // And
+  {a: 1, b: 1, na: 0, nb: 0, less: 0, cin: 0, op: 0b00, out: 1, cout: 0, set: 0 },
+
+  // Or
+  {a: 1, b: 1, na: 0, nb: 0, less: 0, cin: 0, op: 0b01, out: 1, cout: 0, set: 0 },
+  {a: 0, b: 1, na: 0, nb: 0, less: 0, cin: 0, op: 0b01, out: 1, cout: 0, set: 0 },
+  {a: 0, b: 0, na: 0, nb: 0, less: 0, cin: 0, op: 0b01, out: 0, cout: 0, set: 0 },
+
+];
+
+/**
+ * A 1-bit MipsAlu
+ */
+class MipsAlu extends BuiltInGate {
+  /**
+   * Logic:
+   *
+   *   a & b: op = 0
+   *   a | b: op = 1
+   *   a NOR b => ~a & ~b: op = 0, na = 1, nb = 1
+   *
+   * Math:
+   *
+   *   a + b: op = 2
+   *   a - b: op = 2, nb = 1, cin = 1
+   *   b - a: op = 2, na = 1, cin = 1
+   *
+   */
+  eval() {
+    const [ a, b, na, nb, less, cin, op ] = this.getInputPins().map(p => p.getValue());
```

Comment:
nit: in actual code try to maintain ~80 cols, also no space after `[`, and before `]`

```js
const [a, b, na, nb, less, cin, op] =
  this.getInputPins().map(pin => pin.getValue())
```